.github/workflows/release.yml

```diff
@@ -31,14 +31,26 @@ jobs:
         uses: actions/setup-go@d35c59abb061a4a6fb18e82ac0862c26744d6ab5 # pin@5.5.0
         with:
           go-version: '${{ env.GOLANG_VERSION }}'
+
       - name: Run unit tests
         run: make test-coverage
+
       - name: Build Helm Binaries
         run: |
           set -eu -o pipefail
 
-          make build-cross VERSION="${{ github.ref_name }}"
-          make dist checksum VERSION="${{ github.ref_name }}"
+      - name: Run Go Release
+        uses: goreleaser/goreleaser-action@v6.3.0
+        if: success() && startsWith(github.ref, 'refs/tags/')
```

Comment:
I this necessary? given https://github.com/helm/helm/pull/30913/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34L19